app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -126,6 +127,18 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
         super.onViewCreated(view, savedInstanceState)
         consumeFrom(bookmarkStore) {
             bookmarkView.update(it)
+
+            // Only display the sign-in prompt if we're inside of the virtual "Desktop Bookmarks" node.
+            // Don't want to pester user too much with it, and if there are lots of bookmarks present,
+            // it'll just get visually lost. Inside of the "Desktop Bookmarks" node, it'll nicely stand-out,
+            // since there are always only three other items in there. It's also the right place contextually.
+            if (it.tree?.guid == BookmarkRoot.Root.id &&
+                requireComponents.backgroundServices.accountManager.authenticatedAccount() == null
+            ) {
+                signInView.view.visibility = View.VISIBLE
+            } else {
+                signInView.view.visibility = View.GONE
+            }
```

Comment:
nit: is `isVisible` here.

```suggestion
            signInView.view.isVisible = it.tree?.guid == BookmarkRoot.Root.id &&
                requireComponents.backgroundServices.accountManager.authenticatedAccount() == null
```

This would probably be easier to read if we get `accountManager` outside of `consumeFrom`:
```kt
super.onViewCreated(view, savedInstanceState)
val accountManager = requireComponents.backgroundServices.accountManager
consumeFrom(bookmarkStore) {
	...accountManager..authenticatedAccount() == null
}
```